src/server/plugins/plugin.js

```diff
@@ -127,8 +127,10 @@ module.exports = class Plugin {
         server.exposeStaticDir(`/plugins/${id}/{path*}`, this.publicDir);
       }
 
-      this.status = kbnServer.status.createForPlugin(this);
-      server.expose('status', this.status);
+      if (this.externalInit !== _.noop) {
```

Comment:
Oh, I see. externalInit defaults to _.noop. Thanks for explaining.

On Tuesday, August 9, 2016, Peter Pisljar <notifications@github.com> wrote:

> In src/server/plugins/plugin.js
> <https://github.com/elastic/kibana/pull/7953#discussion_r74188822>:
>
> > @@ -127,8 +127,10 @@ module.exports = class Plugin {
> >          server.exposeStaticDir(`/plugins/${id}/{path*}`, this.publicDir);
> >        }
> >
> > -      this.status = kbnServer.status.createForPlugin(this);
> > -      server.expose('status', this.status);
> > +      if (this.externalInit !== _.noop) {
>
> well... because its never undefined and its always a function ? :)
> this.externalInit will always be a function.
>
>    - _.noop in case its not used ... and
>    - a real function in case init was set.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/elastic/kibana/pull/7953/files/91615cf6bbd1e9e254adf93e596062a49b5773d0#r74188822>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABLmg2eI9PZE4di_Yg42-pDtFP5mLEl_ks5qeWb3gaJpZM4Je86g>
> .
>